### PR TITLE
Fix/bsign cleanup

### PR DESCRIPTION
**Pull request by lkindrat-xmos**

### CODE REVIEW

File: lib_nn/src/c/bsign.c [hunk 1]
```diff
@@ -11,72 +11,62 @@
...
+    memset(zero_point_vect, zero_point, VPU_INT8_EPV * sizeof(zero_point));
 
-    for(int k = 0; k < job_count; k++){
-        jobs[k].length = 0;
-    }
-
-    uint32_t left = (length >> 5) << 5;
+    // decompose length = k * job_count * 32 + p * 32 + r,
+    // where 0 <= p < job_count and 0 <= r < 32
+    int32_t r = length % 32;
```
Comment (andrewstanfordjason) [inline lib_nn/src/c/bsign.c]: is 32 bits_per_int32_t?

**Review by andrewstanfordjason — COMMENTED** [review]
the only comment I have is that there is a lot of 32's around. It might be nice to call that what it is, I assume it's bits per int32_t. But the code looks good

File: lib_nn/src/c/bsign.c [hunk 2]
```diff
@@ -11,72 +11,62 @@
...
-            if(left >= 32){
-                jobs[k].length += 32;
-                left -= 32;
-            } else {
-                jobs[k].length += left;
-                left -= left;
-            }
-        }
-        if(left == 0) break;
+    for(int j = 0; j < job_count; j++)
```
Comment (andrewstanfordjason) [inline lib_nn/src/c/bsign.c]: comparison between an int and an unsigned

File: lib_nn/src/c/bsign.c (hunk 1 above)
Comment (lkindrat-xmos) [reply lib_nn/src/c/bsign.c]: That or `VPU_INT8_EPV`. The op only works if the epv is equal to the bitpacked type length in bits.

File: lib_nn/src/c/bsign.c (hunk 2 above)
Comment (lkindrat-xmos) [reply lib_nn/src/c/bsign.c]: I will fix that

**Review by andrewstanfordjason — APPROVED** [review]

**Pull request merged**